lib/puppet/ssl/host.rb

```diff
@@ -559,13 +559,18 @@ def download_and_save_crl_bundle(store=nil)
       # If no SSL store was suppoed, use this host's SSL store
       store ||= ssl_store
       client = http_client(Puppet::Rest::SSLContext.new(OpenSSL::SSL::VERIFY_PEER, store))
+      crl_last_modified = Puppet::FileSystem.exist?(crl_path) ? File.mtime(crl_path) : nil
       Puppet::Util.replace_file(crl_path, 0644) do |file|
-        Puppet::Rest::Routes.get_crls(client, CA_NAME) do |chunk|
+        Puppet::Rest::Routes.get_crls(client, CA_NAME, crl_last_modified) do |chunk|
           file.write(chunk)
         end
       end
     rescue Puppet::Rest::ResponseError => e
-      raise Puppet::Error, _('Could not download CRLs: %{message}') % { message: e.message }
+      if e.response.status_code == 304 # if the ca_crl on master is not newer than the agent's, 304 will be returned.
+        nil
```

Comment:
This branch isn't necessary, does it improve the readability?
Regardless, I think a comment of this length and/or importance should be above the code rather than on the end of the line

Reply:
Fixed it.  Thanks